pkg/endpoint/policy.go

```diff
@@ -498,6 +499,230 @@ func (e *Endpoint) IngressOrEgressIsEnforced() bool {
 		e.Opts.IsEnabled(OptionEgressPolicy)
 }
 
+func (e *Endpoint) GetPortNumber(port string) (uint16, error) {
+	i, err := strconv.Atoi(port)
+	if err != nil {
+		// FIXME check for ports names here if it doesn't exist, the throw initial error
+		return 0, err
+	}
+	return uint16(i), nil
+}
+
+func (e *Endpoint) regeneratePolicyNew(owner Owner, opts models.ConfigurationMap) (*policy.Policy, *policy.Policy, uint64) {
+	policyRepo := owner.GetPolicyRepository()
+
+	oldIngressPolicy := e.IngressPolicy
+	oldEgressPolicy := e.EgressPolicy
+	oldPolicyRev := e.policyRevision
+
+	ingressPolicy := policy.NewPolicy()
+	egressPolicy := policy.NewPolicy()
+
+	ctx := &policy.SearchContext{
+		To: e.SecurityIdentity.Labels.LabelArray(),
+	}
+
+	policyRepo.Mutex.RLock()
+	defer policyRepo.Mutex.RUnlock()
+	ing, egress, rev := policyRepo.ResolvePolicy(ctx)
+
+	labelsMap, _ := getLabelsMap()
+
+	// Create ingress policy
+	for _, ig := range ing {
+		// FIXME needs validation of a singe peer on each ingress rule
+		// Accept only FromEntities or FromCIDR or FromEndpoints or ...
+
+		// Create the L4Port first
+		l4Ports := policy.NewL4RuleContexts()
```

Comment:
One l4ports for everything handles the l4-only case, but not the label-dependent l4 case right?

Reply:
Oh I forgot! This code actually handles for `label-dependent l4` since the l4Ports are used later, check line 564:
```
ingressPolicy.ByNumericIdentity[srcID] = l4Ports
```

For l4-only there will be a special case. When there's no `FromEntities` nor `FromCIDR` nor `FromEndpoints` it will going to be a special entry `ig[numeric.Identity(0)] = l4Ports`